src/cpp/session/modules/SessionFind.cpp

```diff
@@ -487,6 +489,45 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
       }
    }
 
+// permissions getter/setter (only applicable to Unix platforms)
+#ifndef _WIN32
+   Error setPermissions(const std::string& filePath, boost::filesystem::perms permissions)
+   {
+      boost::filesystem::path path(filePath);
+      try
+      {
+         boost::filesystem::permissions(path, permissions);
+      }
+      catch (const boost::filesystem::filesystem_error& e)
+      {
+         return Error(e.code(), ERROR_LOCATION);
+      }
+
+      return Success();
+   }
+
+   Error getPermissions(const std::string& filePath, boost::filesystem::perms* pPerms)
+   {
+      *pPerms = boost::filesystem::no_perms;
+
+      boost::filesystem::path path(filePath);
+      try
+      {
+         boost::filesystem::file_status fileStatus = status(path);
+         *pPerms = fileStatus.permissions();
+         return Success();
```

Comment:
Nit: I think this return should be outside the try block for clarity (as it is in the function above)

Reply:
Done!